src/vm/script.rs

```diff
@@ -322,6 +322,129 @@ impl Script {
                             }
                         }
 
+                        ScriptExecutorState::ExpectingRandomTerm(OP::RandomHash160Var) => {
+                            frame.stack.push(VmTerm::Hash160(rng.gen::<[u8; 20]>()));
+                            frame.executor.state = ScriptExecutorState::ExpectingInitialOP;
+                            frame.i_ptr += 1;
+                            memory_size += 20;
+                        }
+
+                        ScriptExecutorState::ExpectingRandomTerm(OP::RandomHash256Var) => {
+                            frame.stack.push(VmTerm::Hash256(rng.gen::<[u8; 32]>()));
+                            frame.executor.state = ScriptExecutorState::ExpectingInitialOP;
+                            frame.i_ptr += 1;
+                            memory_size += 32;
+                        }
+
+                        // ScriptExecutorState::ExpectingRandomTerm(OP::RandomHash512Var) => {
```

Comment:
Here you can generate two `[u8; 32]` and then use `copy_from_slice`: https://doc.rust-lang.org/std/primitive.slice.html#method.copy_from_slice